apps/multiplatform/common/src/commonMain/kotlin/chat/simplex/common/App.kt

```diff
@@ -194,24 +193,27 @@ fun AndroidScreen(settingsState: SettingsViewState) {
       StartPartOfScreen(settingsState)
     }
     val scope = rememberCoroutineScope()
-    val onComposed: () -> Unit = {
+    val onComposed: suspend (chatId: String?) -> Unit = { chatId ->
+      // coroutine, scope and join() because:
+      // - it should be run from coroutine to wait until this function finishes
+      // - without using scope.launch it throws CancellationException when changing user
+      // - join allows to wait until completion
       scope.launch {
         offset.animateTo(
-          if (chatModel.chatId.value == null) 0f else maxWidth.value,
+          if (chatId == null) 0f else maxWidth.value,
           chatListAnimationSpec()
         )
-        if (offset.value == 0f) {
-          currentChatId = null
-        }
-      }
+      }.join()
```

Comment:
what does it do?

Reply:
There is a comment describing why it's here:
https://github.com/simplex-chat/simplex-chat/pull/3027/files/198873ecad601c0acfba8f8bf3c7aaa274cb54e8#diff-ac9b6f94f88dca783c51ee54710c5c043ed44b7c9a7e49e8187e9b26003a378cR197-R200